File: web/src/types/watchlist.ts
/** 自选基金项 */
export interface WatchlistItem {
  id: number
  fund_code: string
  fund_name: string | null
  group_name: string
  notes: string | null
  added_at: string | null
}

/** 创建自选基金请求 */
export interface WatchlistCreateRequest {
  fund_code: string
  fund_name?: string
  group_name?: string
  notes?: string
}

File: src/fund_analyzer/api/schemas/watchlist.py
"""Watchlist API 请求/响应模型。"""
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel


class WatchlistItem(BaseModel):
    """自选基金条目。"""

    id: int
    fund_code: str
    fund_name: str | None = None
    group_name: str
    notes: str | None = None
    added_at: datetime | None = None


class WatchlistCreateRequest(BaseModel):
    """添加自选请求。"""

    fund_code: str
    fund_name: str = ""
    group_name: str = "默认"
    notes: str = ""
